junebug/tests/helpers.py

```diff
@@ -143,15 +147,24 @@ def create_channel(
         '''Creates and starts, and saves a channel, with a
         TelnetServerTransport transport'''
         if transport_class is None:
-            transport_class = TelnetServerTransport
+            transport_class = 'vumi.transports.telnet.TelnetServerTransport'
 
         properties = deepcopy(properties)
         if config is None:
-            config = yield self.create_channel_config()
+            config = yield self.create_channel_config(
+                channels={
+                    'test_channel': transport_class
+                })
+
+        patched_type = properties['type']
+        properties['type'] = 'test_channel'
         channel = Channel(
             redis, config, properties, id=id, plugins=plugins)
-        properties['config']['transport_name'] = channel.id
         yield channel.start(self.service)
+        properties['type'] = patched_type
```

Comment:
We patch the `type` because we want the channel configuration (returned when `GET`ing the channel) to reflect a valid type, e.g. it should return `telnet` not `test_channel`.